gems/pending/disk/modules/VhdxDisk.rb

```diff
@@ -0,0 +1,634 @@
+    post_process_metadata
+  end
+
+  def post_process_metadata
+    @chunk_ratio                = (2**23 * @logical_sector_size) / @payload_block_size
+    @data_blocks_count          = (@virtual_disk_size / @payload_block_size).ceil
+    @sector_bitmap_blocks_count = (@data_blocks_count / @chunk_ratio).ceil
+    @sectors_per_block          = @payload_block_size / @logical_sector_size
+    if @has_parent
+      @total_bat_entries = @sector_bitmap_blocks_count * (@chunk_ratio + 1)
+    else
+      @total_bat_entries = @data_blocks_count + ((@data_blocks_count - 1) / @chunk_ratio).floor
+    end
+  end
+
+  def process_metadata_table_entry(entry_number)
+    entry_offset = @metadata_offset + SIZEOF_VHDX_METADATA_TABLE_HEADER +
+                   (entry_number - 1) * SIZEOF_VHDX_METADATA_TABLE_ENTRY
+    @vhdx_file.seek(entry_offset, IO::SEEK_SET)
+    metadata_table_entry = VHDX_METADATA_TABLE_ENTRY.decode(@vhdx_file.read(SIZEOF_VHDX_METADATA_TABLE_ENTRY))
+    guid             = metadata_table_entry['item_id']
+    offset           = @metadata_offset + metadata_table_entry['offset']
+    length           = metadata_table_entry['length']
+    bits             = metadata_table_entry['bit_fields']
+    @is_user         = bits[0].to_i(2)
+    @is_virtual_disk = bits[1].to_i(2)
+    @is_required     = bits[2].to_i(2)
+    if length == 0
+      raise "Invalid Metadata Table Entry - Length Zero and Offset is #{offset}" if offset != 0
+      $log.debug "Metadata Table Entry Present but Empty"
+      return
+    end
+    @vhdx_file.seek(offset, IO::SEEK_SET)
+    guid_found = nil
+    METADATA_OPS.each do |meta_guid, op|
+      next unless guid_match?(guid, meta_guid)
+      method(op).call(offset)
+      guid_found = guid
+      break
+    end
+    raise "Invalid Metadata Table Entry GUID #{guid}" unless guid_found
+  end
+
+  def file_parameters(_offset)
+    file_parameters_entry   = VHDX_FILE_PARAMETERS.decode(@vhdx_file.read(SIZEOF_VHDX_FILE_PARAMETERS))
+    @payload_block_size     = file_parameters_entry['block_size']
+    bits                    = file_parameters_entry['bit_fields']
+    @leave_blocks_allocated = bits[0].to_i(2) == 1 ? true : false
+    @has_parent             = bits[1].to_i(2) == 1 ? true : false
+    $log.debug "Payload Block Size #{@payload_block_size} Leave Blocks Allocated #{@leave_blocks_allocated}"
+    $log.debug "Has Parent #{@has_parent}"
+  end
+
+  def virtual_disk_size(_offset)
+    virtual_disk_metadata = VHDX_VIRTUAL_DISK_SIZE.decode(@vhdx_file.read(SIZEOF_VHDX_VIRTUAL_DISK_SIZE))
+    @virtual_disk_size    = virtual_disk_metadata['virtual_disk_size']
+    $log.debug "Virtual Disk Size #{@virtual_disk_size}"
+  end
+
+  def page_83_data(_offset)
+    page_83_metadata = VHDX_PAGE_83_DATA.decode(@vhdx_file.read(SIZEOF_VHDX_PAGE_83_DATA))
+    @page_83_data    = page_83_metadata['page_83_data']
+    $log.debug "Page 83 Data #{@page_83_data}"
```

Comment:
@jrafanie I'd be interested to see your data since I can only find about half that number of calls.  I did find one case of err.backtrace.join(*\n") in a log.debug call that could be enclosed in brackets but all the others are cases of simple variable substitution.  Also note that of the 24 log.debug calls using err.backtrace.join in the entire source tree including this PR only 5 are enclosed in brackets so if this is indeed an issue I can add a PR to rectify the situation.  Let me know if this is necessary.